src/main/java/com/urswolfer/gerrit/client/rest/http/groups/GroupsRestClient.java

```diff
@@ -0,0 +1,117 @@
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Extend the Gerrit implementation in order to add custom functionality
+ * that is not currently supported in the API.
+ *
+ * @author Shawn Stafford
+ */
+public class GroupsRestClient implements Groups {
+
+    private final GerritRestClient gerritRestClient;
+    private final GroupsParser groupsParser;
+
+    public GroupsRestClient(GerritRestClient gerritRestClient, GroupsParser groupsParser) {
+        this.gerritRestClient = gerritRestClient;
+        this.groupsParser = groupsParser;
+    }
+
+    private String getRequestPath() {
+        return "/groups";
+    }
+
+    /**
+     * Look up a group by ID.
+     * <p>
+     * <strong>Note:</strong> This method eagerly reads the group. Methods that
+     * mutate the group do not necessarily re-read the group. Therefore, calling a
+     * getter method on an instance after calling a mutation method on that same
+     * instance is not guaranteed to reflect the mutation. It is not recommended
+     * to store references to {@code groupApi} instances.
+     *
+     * @param id any identifier supported by the REST API, including group name or
+     *     UUID.
+     * @return API for accessing the group.
+     * @throws RestApiException if an error occurred.
+     */
+    public GroupApi id(String id) throws RestApiException {
+        return new GroupApiRestClient(gerritRestClient, groupsParser, id);
+    }
+
+    /** Create a new group with the given name and default options. */
+    public GroupApi create(String name) throws RestApiException {
+        String restPath = GroupApiRestClient.BASE_URL + "/" + name;
+        JsonElement result = gerritRestClient.putRequest(restPath);
+        GroupInfo info = groupsParser.parseGroupInfo(result);
+        return new GroupApiRestClient(gerritRestClient, groupsParser, info.id);
+    }
+
+    /** Create a new group. */
+    public GroupApi create(GroupInput input) throws RestApiException {
+        String restPath = GroupApiRestClient.BASE_URL + "/" + input.name;
+        String body = gerritRestClient.getGson().toJson(input);
+        JsonElement result = gerritRestClient.putRequest(restPath, body);
+        GroupInfo info = groupsParser.parseGroupInfo(result);
+        return new GroupApiRestClient(gerritRestClient, groupsParser, info.id);
+    }
+
+    /** @return new request for listing groups. */
+    public ListRequest list() {
+        return new ListRequest() {
+            @Override
+            public SortedMap<String, GroupInfo> getAsMap() throws RestApiException {
+                SortedMap<String, GroupInfo> map = new TreeMap<String, GroupInfo>();
+                List<GroupInfo> list = GroupsRestClient.this.list(this);
+                if (list != null) {
+                    Iterator listIter = list.iterator();
+                    while (listIter.hasNext()) {
```

Comment:
I prefer a foreach here (for (GroupInfo group : list) {...), but it's up to you :)